pkg/security/mcs/label.go

```diff
@@ -0,0 +1,178 @@
+package mcs
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+const maxCategories = 1024
+
+type Labels struct {
```

Comment:
An selinux label consists of 3-5 components.  An example label is `system_u:system_r:cupsd_t:s0-s0:c0.c1023`

The components are `user:role:type:levels:categories`   the last 2 components together (`levels:categories`) are collectively  also known as the `range`.  So in the above example

| SELinux Term | Example Value  |
| --- |---|
| user | systemd_u |
| role |  systemd_r |
| type | cupsd_t |
| levels | s0-s0 |
| categories | c0.c1023 |

Not all selinux systems support a `range`.  Some systems ONLY support the user:role:type portions.  MCS (the mechanism used to separate permissions between two docker containers) is based almost entirely on the `range`.


There are probably only a handful of people in the world who really use the `levels` for anything meaningful.  Ignore them.  Just use `s0`.

So code here, following what was done in sVirt for libvirt or in docker, we only need to calculate/mutate the `categories`. Getting the 'correct` type is also going to be important.  I have no reviewed the code however to see what is being done.